File: samples/nodejs/src/encryption/KeyAndIV.ts
import {EncryptionKey} from "./EncryptionKey";
import {InitializationVector} from "./InitializationVector";
import {SerializedKeyAndIV} from "./SerializedKeyAndIV";

export interface KeyAndIV {
    cipherName?: string
    key?: EncryptionKey
    IV?: InitializationVector

    tagLength?: number

    serialize(): SerializedKeyAndIV;
}

File: samples/java/persistence/src/main/java/io/sytac/encryption/KeyAndIV.java
package io.sytac.encryption;

import java.security.spec.AlgorithmParameterSpec;

/**
 * Key and IV (initialization vector) for AEAD (authenticated encryption with associated data).
 */
public interface KeyAndIV {
    AlgorithmParameterSpec createAlgorithmSpec();

    String getCipherName();

    InitializationVector getIV();

    EncryptionKey getKey();

    int getTagLength();

    SerializedKeyAndIV serialize();
}
